src/validate.js

```diff
@@ -3100,89 +3103,171 @@ function F_VALIDATE(disabledHL) {
+
+			// mark venue as seen
+			_RT.$seen[venueID] = seen = [0, null, false, false,
+				4 > currentZoom,
+				cityID];
+
+			// increase city counter
+			venue.incCityCounter();
+
+			// shortcuts
+			var address = venue.$address;
+			var country = address.$country;
+			var countryLen = country.length;
+			var countryCode = country ? _I18n.getCountryCode(country.toUpperCase())
+				: _RT.$cachedTopCCode;
+			var city = address.$city;
+			var cityLen = city.length;
+			var cityID = address.$cityID;
+			var street = address.$street;
+			var state = address.$state;
+			var streetLen = street.length;
+			var alts = venue.$alts;
+			var lock = venue.$lock;
+
+			var exceptedCategories ='BRIDGE|ISLAND|FOREST_GROVE|SEA_LAKE_POOL|RIVER_STREAM|CANAL|DAM|TUNNEL|JUNCTION_INTERCHANGE'.split('|');
+
+			if (!cityLen
+				&& exceptedCategories.indexOf(venue.$categories[0]) === -1
+				&& isLimitOk(250)
+				&& address.isOkFor(250))
+				venue.report(250);
+
+			// GROUP name.length
+			if (!venue.$name.length) {
+				if (venue.$categories[0] === "PARK" && !cityLen
+					&& isLimitOk(258)
+					&& address.isOkFor(258))
+					venue.report(258);
+			}
+			// GROUP name.length
+			// Check for last update by bots
+			var botNamesAndIDs = [
+				'^waze-maint', '^105774162$',
+				'^waze3rdparty$', '^361008095$',
+				'^WazeParking1$', '^338475699$',
+				'^admin$', '^-1$',
+				'^avsus$', '^107668852$'
+			];
+			var re = new RegExp(botNamesAndIDs.join('|'),'i');
+
+			if ((re.test(venue.$updatedByID.toString()) || re.test(venue.$updatedBy))
+				&& isLimitOk(251)
+				&& address.isOkFor(251))
+				venue.report(251);
+
+			// GROUP isParkingLot
+			if (venue.$rawObject.isParkingLot()){
+				var catAttr = venue.$categoryAttributes;
+				var parkAttr = catAttr ? catAttr.PARKING_LOT : undefined;
+				// missing parking lot type
+				if ((!parkAttr || !parkAttr.parkingType)
+					&& isLimitOk(252)
+					&& address.isOkFor(252))
+					venue.report(252);
+				// missing cost type
+				if ((!parkAttr || !parkAttr.costType || parkAttr.costType === 'UNKNOWN')
+					&& isLimitOk(253)
+					&& address.isOkFor(253))
+					venue.report(253);
+				// missing payment types
+				if ((parkAttr && parkAttr.costType && parkAttr.costType !== 'FREE'
+					&& parkAttr.costType !== 'UNKNOWN'
+					&& (!parkAttr.paymentType || !parkAttr.paymentType.length))
+					&& isLimitOk(254)
+					&& address.isOkFor(254))
+					venue.report(254);
+				//check elevation of the parking lot
+				if ((!parkAttr || !parkAttr.lotType || parkAttr.lotType.length === 0)
+					&& isLimitOk(255)
+					&& address.isOkFor(255))
+					venue.report(255)
+				if ((!venue.$entryExitPoints || !venue.$entryExitPoints.length)
+					&& isLimitOk(257)
+					&& address.isOkFor(257))
+					venue.report(257)
+			}// GROUP isParkingLot
+
+			// GROUP isGasStation
+			if (venue.$rawObject.isGasStation()) {
+				// check if brand in name
+				if (venue.$name.indexOf(venue.$brand) === -1
+					&& isLimitOk(259)
```

Comment:
Overall, it is not necessary to use `isLimitOk()`, since the `venue.report()` below will check the limit anyway.

The rationale behind `isLimitOk()` is to avoid long computations. For example:

    if (isLimitOk()) {
        // long check
        if (condition)
            report();
   }

In such cases it makes sense, since if the limit is exceeded, we do not even start the "long check" part.

Reply:
Ah, okay, that makes sense. I need to review the checks then. Thanks for the explanation.